presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -583,6 +583,25 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
   return str;
 }
 
+std::string PrestoTask::toString() const {
+  // Can this be in a deadlock due to a bug?
+  std::lock_guard<std::mutex> l(mutex);
+
+  std::ostringstream oss;
+  oss << "{PrestoTask: ";
```

Comment:
Let's use folly::dynamic to create a dynamic type and return json from there. String manipulation for json is not good.

https://github.com/facebook/folly/blob/main/folly/docs/Dynamic.md

Since the toString() method is returning a `json` string, shall we name it toJsonString()?

Reply:
Done